### DISCUSSION TITLE
T95MAX+

### DISCUSSION THREAD

**Discussion by vpm** [devmfc/debian-on-amlogic#46] [General]
Hi
You can add T95MAX+ with RAM 4GB and emmc 64GB from Aliexpress
config box=s905x3_generic
ethernet 100m working
wifi working, show as broadcom 4330, but real chip is Sp6330
bluetouth not working
hdmi ok
sounds not tested

RAM only show 3.2 GB, but when i install armbian i have 3.8 GB RAM. I try get dtb-file from armbian but whithout changes - i see only 3.2GB

lshw 
tvbox                       
    description: Computer
    product: Meson SM1 ac2xx reference board minimal 100M
    width: 64 bits
    capabilities: smp tagged_addr_disabled
  *-core
       description: Motherboard
       physical id: 0
     *-cpu:0
          description: CPU
          product: cpu
          physical id: 0
          bus info: cpu@0
          size: 1908MHz
          capacity: 1908MHz
          capabilities: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp cpufreq
     *-cpu:1
          description: CPU
          product: cpu
          physical id: 1
          bus info: cpu@1
          size: 1908MHz
          capacity: 1908MHz
          capabilities: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp cpufreq
     *-cpu:2
          description: CPU
          product: cpu
          physical id: 2
          bus info: cpu@2
          size: 1908MHz
          capacity: 1908MHz
          capabilities: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp cpufreq
     *-cpu:3
          description: CPU
          product: cpu
          physical id: 3
          bus info: cpu@3
          size: 1908MHz
          capacity: 1908MHz
          capabilities: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp cpufreq
     *-cpu:4 DISABLED
          description: CPU
          product: l2-cache0
          physical id: 4
          bus info: cpu@4
     *-memory
          description: System memory
          physical id: 5
          size: 3267MiB
  *-mmc0
       description: MMC Host
       physical id: 1
       logical name: mmc0
  *-mmc1
       description: MMC Host
       physical id: 2
       logical name: mmc1
     *-device
          description: SD/MMC Device
          product: CJNB4R
          vendor: Unknown (21)
          physical id: 1
          bus info: mmc@1:0001
          date: 01/2016
          serial: 4267174621
          capabilities: mmc
        *-interface:0
             physical id: 1
             logical name: /dev/mmcblk1rpmb
        *-interface:1
             physical id: 2
             logical name: /dev/mmcblk1
             size: 62537072640
             capabilities: partitioned partitioned:dos
             configuration: logicalsectorsize=512 sectorsize=512 signature=2e330000
           *-volume:0
                description: Windows FAT volume
                vendor: mkfs.fat
                physical id: 1
                logical name: /dev/mmcblk1p1
                logical name: /boot
                version: FAT32
                serial: d52b-7c3f
                size: 489MiB
                capacity: 512MiB
                capabilities: primary fat initialized
                configuration: FATs=2 filesystem=fat label=BOOT mount.fstype=vfat mount.options=rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,utf8,errors=remount-ro state=mounted
           *-volume:1
                description: EXT4 volume
                vendor: Linux
                physical id: 2
                logical name: /dev/mmcblk1p2
                version: 1.0
                serial: b81c1ceb-55ad-4018-ab3e-86165782bc33
                size: 56GiB
                capacity: 56GiB
                capabilities: primary journaled extended_attributes large_files huge_files dir_nlink recover 64bit extents ext4 ext2 initialized
                configuration: created=2024-03-29 19:49:54 filesystem=ext4 label=ROOTFS lastmountpoint=/ modified=1970-01-01 01:00:01 mounted=1970-01-01 01:00:01 state=clean
  *-mmc2
       description: MMC Host
       physical id: 3
       logical name: mmc2
     *-device
          description: SDIO Device
          physical id: 1
          bus info: mmc@2:0001
          serial: 0
          capabilities: sdio
        *-interface:0
             description: Wireless interface
             product: 4330
             vendor: Broadcom
             physical id: 1
             bus info: mmc@2:0001:1
             logical name: mmc2:0001:1
             logical name: wlan0
             serial: e0:76:d0:2c:5e:68
             capabilities: ethernet physical wireless
             configuration: broadcast=yes driver=brcmfmac driverversion=5.90.125.104 firmware=5.90.125.104 multicast=yes wireless=IEEE 802.11
        *-interface:1
             product: 4330
             vendor: Broadcom
             physical id: 2
             bus info: mmc@2:0001:2
             logical name: mmc2:0001:2
  *-sound
       description: MESONSM1
       physical id: 4
       logical name: card0
       logical name: /dev/snd/controlC0
       logical name: /dev/snd/pcmC0D0p
       logical name: /dev/snd/pcmC0D1p
       logical name: /dev/snd/pcmC0D2p
       logical name: /dev/snd/pcmC0D3c
       logical name: /dev/snd/pcmC0D4c
       logical name: /dev/snd/pcmC0D5c
  *-graphics
       product: mesondrmfb
       physical id: 5
       logical name: /dev/fb0
       capabilities: fb
       configuration: depth=32 resolution=1024,768
  *-input
       product: meson-ir
       physical id: 6
       logical name: input0
       logical name: /dev/input/event0
       capabilities: platform
  *-network
       description: Ethernet interface
       physical id: 7
       logical name: eth0
       serial: c4:2a:fe:83:60:42
       size: 100Mbit/s
       capacity: 100Mbit/s
       capabilities: ethernet physical tp mii 10bt 10bt-fd 100bt 100bt-fd autonegotiation
       configuration: autonegotiation=on broadcast=yes driver=st_gmac driverversion=6.6.23-meson64 duplex=full ip=192.168.88.112 link=yes multicast=yes port=twisted pair speed=100Mbit/s

**Comment by devmfc** [comment]
That is good to hear, thanks for reporting back.

You could try the Tanix TX3 100M box config, I might just make bluetooth work as well.

**Reply by gitisgreat2023** [reply]
@devmfc 
Interesting, he also reports that only 3.2 GB is detected... I tried your newest release.
Do you have a 4GB box where the full amount is available?

`RAM only show 3.2 GB, but when i install armbian i have 3.8 GB RAM. I try get dtb-file from armbian but whithout changes - i see only 3.2GB`

**Reply by devmfc** [reply]
None of my S905X3 boxes show more than 3.4GiB available. AFAIK that is normal. Although I admit I think the difference between 4GiB and 3.2 GiB is quite large.

**Comment by gitisgreat2023** [comment]
What shows cat /proc/meminfo 
MemTotal:        

My other armbian box 4002440 kB

**Reply by devmfc** [reply]
`cat /proc/iomem`

and you will see the reason I think.
I leave the memory node out of my dtb's and trust the bootloaders to do the right thing. (they set the right memory node and resrverd memory nodes)

Your other box is not mainline kernel? Which image do you use?

~~~
#cat /proc/meminfo
MemTotal:        3345864 kB
MemFree:         2864804 kB
MemAvailable:    3161924 kB
Buffers:            9100 kB
Cached:           343528 kB
SwapCached:            0 kB
Active:            56232 kB
Inactive:         322768 kB
Active(anon):        632 kB
Inactive(anon):    28264 kB
Active(file):      55600 kB
Inactive(file):   294504 kB
Unevictable:          16 kB
Mlocked:              16 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Zswap:                 0 kB
Zswapped:              0 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:         26400 kB
Mapped:            39396 kB
Shmem:              2524 kB
KReclaimable:      18188 kB
Slab:              49488 kB
SReclaimable:      18188 kB
SUnreclaim:        31300 kB
KernelStack:        2048 kB
PageTables:         1836 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     1672932 kB
Committed_AS:     118380 kB
VmallocTotal:   133141626880 kB
VmallocUsed:        7096 kB
VmallocChunk:          0 kB
Percpu:             1056 kB
EarlyMemtestBad:       0 kB
HardwareCorrupted:     0 kB
AnonHugePages:         0 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
CmaTotal:         262144 kB
CmaFree:          245068 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB

~~~

**Reply by gitisgreat2023** [reply]
other box is a standard armbian, 6.6.2-edge-rockchip64

Okay, at the X3 I noticed "Reserved memory: created CMA memory pool at 0x00000000c3c00000, size 256 MiB". 
Yours also shows it: `CmaTotal:         262144 kB`

OK, digged further.... see [this](https://forum.armbian.com/topic/36945-64-mb-less-total-memory-on-2421/). 

The 6.6.y kernel for reasons I have yet to figure out reads the MEM wrong. 6.7 and up is not affected by this. Nor is 6.1 to 6.5.

How do I upgrade to a newer kernel, if available?

**Reply by devmfc** [reply]
Rockchip and Amlogic are quite different...

You could try downgrading the kernel to 6.1 to test if it makes a difference, but I doubt it...

**Reply by gitisgreat2023** [reply]
Yeah that's right.
How do I upgrade/downgrade the kernel for your release?

**Reply by devmfc** [reply]
dpkg -i linux-image-xyz.deb

**Reply by claudiologiudice** [reply]
Have you tested the vfd display?

**Comment by ghost** [comment]
latest kernel [v6.12.6, not the .5] have boot loop in my headless unit.... revert back to 6.6.60 is good solution for now...btw i'll stay with this kernel 'couse is stable and work well... (using it as mysql server + nfs sharing server for 4 kodi client) and is absolutely great! :D

**Reply by devmfc** [reply]
Sorry to hear that... That's why I only do a major kernel update every year....
Do you have some output or logs? Which DTB or box config do you use? 
You can of course stick with the 6.6 kernel for now, but there will be no more updates to that kernel from my side, so at some point it will get insecure... 
That's why I'm very keen to solve the problems with the 6.12 kernel, to give the same stability for the coming year. But I really need logs and as much information as you can give for that...

**Reply by ghost** [reply]
i think my issue is related to fstab and automount of my 2 hdd...'couse i remember i'll try to disconnect hdd and the box boot up. btw, i use tanix3 config. if is needed, maybe i can try to upgrade now and see what happen! really not a big issue try this.. Only need to backup some DB and try... 

edit.
update.
i try!

**Comment by ghost** [comment]
![Screenshot 2024-12-30 121154](https://github.com/user-attachments/assets/0779295e-a603-4c95-bf0f-3ed86ca7e6fe)

**Comment by ghost** [comment]
here log after update...
at this moment is up and running.
i try to see what happen.

[-- Boot 41b545d7d9614d9ebe908477c84.txt](https://github.com/user-attachments/files/18274250/--.Boot.41b545d7d9614d9ebe908477c84.txt)

**Comment by ghost** [comment]
![Screenshot 2024-12-31 054223](https://github.com/user-attachments/assets/e4e5442f-6796-4094-8651-c650b693c882)


ok, 17,5H up and running without issue. 
I send now a reboot command, up and running in 'bout 4/5sec

ALL PERFECT. i think is absolutely error with fstab and one disk. unmount "fault" (is not fault but onlyprevious  didn't ejected properly), re-mount, reboot...upgrade, test. 

Seems all good and working like a charm ;-) 

Think issue solved.
Tnx for interesting :-)

**Reply by devmfc** [reply]
Super, good to hear!
